taskweaver/role/translator.py

```diff
@@ -262,66 +256,34 @@ def parse_llm_output_stream_v2(
-        cur_content: Optional[str] = None
-
         try:
             for ev in parser:
-                if ev.prefix == root_element_prefix:
-                    if ev.event == "start_array":
-                        list_begin = True
-                    if ev.event == "end_array":
-                        list_end = True
-
-                if not list_begin or list_end:
-                    continue
-
-                cur_item_prefix = f"{root_element_prefix}[{item_idx}]"
-                if ev.prefix == cur_item_prefix:
-                    if ev.event == "start_map":
-                        cur_content_sent, cur_content_sent_end = False, False
-                        cur_type, cur_content = None, None
-                    if ev.event == "end_map":
-                        if cur_type is None or cur_content is None:
-                            raise Exception(
-                                f"Incomplete generate kv pair in index {item_idx}. "
-                                f"type: {cur_type} content {cur_content}",
-                            )
-
-                        if cur_content_sent and not cur_content_sent_end:
-                            # possible incomplete string, trigger end prematurely
-                            yield cur_type, "", True
-
-                        if not cur_content_sent:
-                            yield cur_type, cur_content, True
-
-                        cur_content_sent, cur_content_sent_end = False, False
-                        cur_type, cur_content = None, None
-                        item_idx += 1
-
-                if ev.prefix == cur_item_prefix + ".type":
-                    if ev.event == "string" and ev.is_end:
-                        cur_type = ev.value
-
-                if ev.prefix == cur_item_prefix + ".content":
-                    if ev.event == "string":
-                        if cur_type is not None:
-                            cur_content_sent = True
-                            yield cur_type, ev.value_str, ev.is_end
-
-                            assert not cur_content_sent_end, "Invalid state: already sent is_end marker"
-                            if ev.is_end:
-                                cur_content_sent_end = True
-                        if ev.is_end:
-                            cur_content = ev.value
+                if ev.prefix == root_element_prefix and ev.event == "map_key" and ev.is_end:
+                    cur_type = ev.value
+                    yield cur_type, "", False
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "string":
+                    yield cur_type, ev.value_str, ev.is_end
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "number":
+                    yield cur_type, ev.value_str, ev.is_end
+                elif ev.prefix == f"{root_element_prefix}.{cur_type}" and ev.event == "boolean":
+                    yield cur_type, ev.value_str, ev.is_end
```

Comment:
maybe we could also add event type `null` in case some generation would be `{"key1": null, "key2": null}` in parsing some properties and we can terminate the previous key wwith `yield cur_type, "", True` to indicate its end?

Reply:
Will do